bitcoin/src/taproot.rs

```diff
@@ -700,48 +814,184 @@ impl NodeInfo {
     }
 }
 
+impl TryFrom<TaprootBuilder> for NodeInfo {
+    type Error = IncompleteBuilder;
+
+    fn try_from(builder: TaprootBuilder) -> Result<Self, Self::Error> {
+        builder.try_into_node_info()
+    }
+}
+
+#[cfg(feature = "serde")]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
+impl serde::Serialize for NodeInfo {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use serde::ser::SerializeSeq;
+        let mut seq = serializer.serialize_seq(Some(self.leaves.len()))?;
```

Comment:
I think this needs to be multiplied by 2 since there are two calls of `serialize_element`.

Reply:
good catch